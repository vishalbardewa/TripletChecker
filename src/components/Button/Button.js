import React from 'react';
import PropTypes from 'prop-types';

function Button({title, onClick}) {
  return (
      <div className="custom-button">
        <button data-testid='button' onClick={onClick}>{title}</button>
      </div>
  )
}

Button.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func
};

export default Button
