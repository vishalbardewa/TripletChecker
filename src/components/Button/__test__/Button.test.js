import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Button from '../Button';

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div)
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button title="Click Me" />)
  expect(getByTestId('button').textContent).toBe("Click Me")
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button title="Save" />)
  expect(getByTestId('button').textContent).toBe("Save")
})

it("matches snapshot", () => {
  const snap = renderer.create(<Button title="Click Button" />).toJSON();
  expect(snap).toMatchSnapshot();
})
