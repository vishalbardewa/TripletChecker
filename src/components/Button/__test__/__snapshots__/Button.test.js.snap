// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="custom-button"
>
  <button
    data-testid="button"
  >
    Click Button
  </button>
</div>
`;
